TACC-Cloud/hazmapper
# Task/WG-196: handle various environments

## Overview: ##

This PR handles various environments (both local and deployed):

* Adds Dockerfile for React image
* Removes Vite build environment approach as we want a single build or docker image to handle all the different environments
* Add [hooks/environment/useAppConfiguration.ts](https://github.com/TACC-Cloud/hazmapper/pull/189/files#diff-42a7c25c917cb7f49d8c6fbfadaa9167451997006a08f2eaee1cedfbb68c1b21) to handle all envionment-based configuration
* Use useAppConfiguration in login and in how requests are constructed
* Add `react/src/secret_local.example.ts` so local development can be configured to use different GEOAPI backend; JWT is now located in `react/src/secret_local.example.ts`

This PR also:

* bumps node in the Github action
* updates one comment in the angular code related to taggit

## PR Status: ##


## Related Jira tickets: ##

* [WG-196](https://tacc-main.atlassian.net/browse/WG-196)

## Testing Steps: ##
1) See updated README.md and do `cp react/src/secret_local.example.ts react/src/secret_local.ts` and ensure that you have added your JWT there.
2) See that hazmapper works with production (i.e. `EnvironmentType.Production` is in `react/src/secret_local.ts`)
3) Update `react/src/secret_local.ts` to use a different `geoapiBackend` like `EnvironmentType.Production`, `EnvironmentType.Staging`, or `EnvironmentType.Dev`.
4) Note that you can use localhost:4200 or hazmapper.local:4200
5) Visit https://hazmapper.tacc.utexas.edu/dev-react/ to confirm that it is working in a deployed environment

## Notes: ##

Added a follow on Jira issue as resources (index.XXXXX.js and index.XXXXXX.css) are first loaded from root before being loaded from the correct path. https://tacc-main.atlassian.net/browse/WG-224
